core/dbt/version.py

```diff
@@ -27,81 +51,175 @@ def get_latest_version(version_url: str = PYPI_VERSION_URL):
-                    compatibility_msg = yellow("Update available!")
-                else:
-                    compatibility_msg = red("Out of date!")
-                plugin_update_msg = (
-                    "  Your version of dbt-{} is out of date! "
-                    "You can find instructions for upgrading here:\n"
-                    "  https://docs.getdbt.com/dbt-cli/install/overview\n\n"
-                ).format(plugin_name)
-            else:
-                compatibility_msg = yellow("No PYPI version available")
-
-        plugin_version_msg += ("  - {}: {} - {}\n" "{}").format(
-            plugin_name, version, compatibility_msg, plugin_update_msg
-        )
+    for name, version, update_msg in _pad_lines(lines, seperator=":"):
+        line_msg = f"  - {name} {version}"
+        if update_msg != "":
+            line_msg += f" - {update_msg}"
+        msg_lines.append(line_msg)
 
-    if latest is None:
-        return (
-            "{}The latest version of dbt could not be determined!\n"
-            "Make sure that the following URL is accessible:\n{}\n\n{}".format(
-                version_msg, PYPI_VERSION_URL, plugin_version_msg
-            )
-        )
+    return msg + "\n".join(msg_lines)
+
+
+def _get_plugins_msg(installed: Type[dbt.semver.VersionSpecifier]) -> str:
+    msg = "Plugins:\n"
+    msg_lines = []
 
-    if installed == latest:
-        return f"{version_msg}{green('Up to date!')}\n\n{plugin_version_msg}"
+    plugins = []
+    plugin_msgs = []
+    for name, version_s in _get_dbt_plugins_info():
+        compatability_msg, update_msg = _get_plugin_msg_info(name, version_s, installed)
+        plugins.append([name, version_s, compatability_msg])
+        plugin_msgs.append(update_msg)
 
-    elif installed > latest:
-        return "{}Your version of dbt is ahead of the latest " "release!\n\n{}".format(
-            version_msg, plugin_version_msg
+    for i, plugin in enumerate(_pad_lines(plugins, seperator=":")):
+        msg_lines.append(_format_single_plugin(plugin, plugin_msgs[i]))
+    return msg + "\n".join(msg_lines)
+
+
+def _get_plugin_msg_info(
+    name: str, version_s: str, core: Type[dbt.semver.VersionSpecifier]
+) -> Tuple[str, str]:
+    plugin = dbt.semver.VersionSpecifier.from_version_string(version_s)
+    latest_plugin = get_latest_version(version_url=get_package_pypi_url(name))
+
+    update_msg = ""
+
+    if plugin.major != core.major and plugin.minor != core.minor:
+        compatibility_msg = red("Not compatible!")
+        update_msg = (
+            f"  Your version of dbt-{name} is not compatible with core!\n"
+            "  You can find instructions for upgrading here:\n"
+            "  https://docs.getdbt.com/dbt-cli/install/overview"
         )
+        return (compatibility_msg, update_msg)
 
-    else:
-        return (
-            "{}Your version of dbt is out of date! "
+    if not latest_plugin:
+        compatibility_msg = yellow("No PYPI version available")
```

Comment:
[Not blocking, cosmetic only] More like the `dbt-core` message above:
```suggestion
        msg = f"The latest version of dbt-{name} could not be determined from {get_package_pypi_url(name)}"
        compatibility_msg = yellow(msg)
```

Reply:
I think it's a bit overkill to have the name of the version twice, as well as having the URL on the same line. This change would also make this the longest message at the end of a line.
```
Plugins:
  - redshift:         1.0.0 - Update available!
  - bigquery:       1.0.0 - Not compatible!
  - postgres:       1.0.0 - Up to date!
  - snowflake:     1.0.0 - Your version of dbt-snowflake could not be determined from https://pypi.org/pypi/dbt-snowflake/json
```
This seems a bit out of place.

Reply:
Good point — in context it doesn't look quite right. My main interest here was the PyPi link, since we're just naively assuming that the PyPi link is `f"https://pypi.org/pypi/dbt-{name}/json"`. What do you think of:
```suggestion
        compatibility_msg = yellow(f"Could not get latest version from {get_package_pypi_url(name)}")
```

Reply:
Not really — it just lets us show our cards, since we're being naive about where we're checking for the link :)

`"No PYPI version available"` feels, to me, like we're being a bit smarter / more thorough than we really are. I'd be fine with `Could not determine latest version`